### DISCUSSION TITLE
Knowledge

### DISCUSSION THREAD

**Discussion by min-naing-k** [staudenmeir/eloquent-json-relations#115] [General]
Sir, this discusstion is just for knowledge.

When I write this to user migrations,
```php
Schema::create('users', function (Blueprint $table) {
    $table->bigIncrements('id');
    $table->json('options');
    $locale_id = DB::connection()->getQueryGrammar()->wrap('options->locale_id');
    $table->unsignedBigInteger('locale_id')->storedAs($locale_id);
    $table->foreign('locale_id')->references('id')->on('locales');
});
```

can I safely change 'options->locale_id' to 'local_id' in the User model?
```php
class User extends Model
{
    use \Staudenmeir\EloquentJsonRelations\HasJsonRelationships;

    public function locale()
    {
        return $this->belongsTo(Locale::class, 'options->locale_id');
    }
}
```

**Comment by staudenmeir** [comment]
Hi @min-naing-k,
You can replace it if you only use the relationship for lazy and eager loading queries (e.g. `$user->locale` or `User::with('locale')->get()`). However, it doesn't work with use cases like `$user->locale()->associate()` because `options->locale_id` doesn't get updated anymore.

**Reply by min-naing-k** [reply]
Got it 🎉. Thanks, sir...
